src/index.js

```diff
@@ -503,49 +537,61 @@ export default class RestClient {
    *
    * @private
    * @param {string} endpoint
+   * @param {any} body
    * @param {Object} context
    * @param {Object} options
-   * @param {any} body
-   * @param {string} resourceKey
    * @return {Promise}
    */
-  async _post(endpoint, context, options, body, resourceKey) {
-    const res = await this._fetch('POST', endpoint, context, options, body);
-    if (res.errors) return res.errors;
-
-    if (res.data) {
-      this._setCache(castArray(res.data), res.headers, context.path, null, resourceKey);
-    }
+  async _post(endpoint, body, context, options) {
+    const method = 'POST';
+    const fetchOptions = { body, headers: new Headers(options.headers), method };
+    const { data } = await this._fetch(method, endpoint, fetchOptions, context, options);
+    return data;
+  }
 
-    return res.data;
+  /**
+   *
+   * @private
+   * @param {Array<Promise>} promises
+   * @param {string} path
+   * @param {string} fetchID
+   * @return {Object}
+   */
+  async _resolveData(promises, path, fetchID) {
+    const data = flatten(await Promise.all(promises)).filter(value => !!value);
+    const { requests } = this._getTracker(path);
+    const metadata = requests[fetchID];
+    return { data, metadata };
   }
 
   /**
    *
    * @private
-   * @param {Object} res
-   * @param {Array<Object>} res.data
-   * @param {Object} res.errors
-   * @param {string|Array<string>} values
+   * @param {string} resourceKey
+   * @param {Array<string>} values
+   * @param {Array<any>} data
    * @param {Object} context
    * @param {string} context.fetchID
-   * @param {string} context.requestID
    * @param {string} context.path
-   * @param {string} resourceKey
+   * @param {Object} res
    * @return {void}
    */
-  async _resolveRequests({ data, errors }, values, { fetchID, requestID, path }, resourceKey) {
-    if (!values.length) return;
-    const { fetching, pending } = this._getTracker(requestID, path);
+  async _resolveRequests(resourceKey, values, data, { fetchID, path }, res) {
+    const { fetching, pending, requests } = this._getTracker(path);
 
     values.forEach((value) => {
       const index = fetching.findIndex(obj => obj.value === value);
       if (index !== -1) fetching.splice(index, 1);
-      const match = errors || data.find(obj => obj[resourceKey] === value);
+      const match = data.find(obj => get(obj, [resourceKey], null) === value);
       if (!pending[value]) return;
 
       for (let i = pending[value].length - 1; i >= 0; i -= 1) {
-        if (pending[value][i].fetchID === fetchID) {
+        if (pending[value][i].batchID === fetchID) {
+          const _fetchID = pending[value][i].fetchID;
+          if (!requests[_fetchID]) requests[_fetchID] = [];
+          const { data: resData, headers, ...otherProps } = res;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "headers" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "resData" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'resData' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AUnusedVariable)